cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -255,12 +263,19 @@ func buildPartitionList(ptable *sfdiskPartitionTable, pv *gadget.LaidOutVolume)
 		if seen[uint64(start)] {
 			continue
 		}
+
+		// Check if the data partition should be expanded
+		size := s.Size
+		if s.Role == gadget.SystemData && expandData && p.StartOffset+s.Size < dl.Size {
+			size = dl.Size - p.StartOffset
```

Comment:
IIRC `last-lba` is the last usable sector, so we could`+ dl.SectorSize` here?. OTOH, it's only a single sector, so there's negligible win here. Also, it'd be worth to check whether sfdisk complains about it first.

Reply:
In this case I believe DeviceLayout.Size should be fixed to use `(ptable.LastLBA + 1) * sectorSize` and the size computation here should remain the same. Will fix it there.